src/nbl/video/ILogicalDevice.cpp

```diff
@@ -20,36 +25,37 @@ class SpirvDebloatTask
         void insertEntryPoint(const IGPUPipelineBase::SShaderSpecInfo& shaderSpec, hlsl::ShaderStage stage)
         {
             const auto* shader = shaderSpec.shader;
-            auto it = m_entryPointsMap.find(shader);
-            if (it == m_entryPointsMap.end() || it->first != shader)
-                it = m_entryPointsMap.emplace_hint(it, shader, EntryPoints());
-            it->second.insert({ .name = shaderSpec.entryPoint, .stage = stage });
+            auto it = m_shaderInfoMap.find(shader);
+            if (it == m_shaderInfoMap.end() || it->first != shader)
+              it = m_shaderInfoMap.emplace_hint(it, shader, ShaderInfo{ EntryPoints(), nullptr } );
+            it->second.entryPoints.insert({ .name = shaderSpec.entryPoint, .stage = stage });
         }
 
         IGPUPipelineBase::SShaderSpecInfo debloat(const IGPUPipelineBase::SShaderSpecInfo& shaderSpec, core::vector<core::smart_refctd_ptr<const asset::IShader>>& outShaders)
         {
             const auto* shader = shaderSpec.shader;
-            const auto& entryPoints = m_entryPointsMap[shader];
+            auto findResult = m_shaderInfoMap.find(shader);
+            assert(findResult != m_shaderInfoMap.end());
+            const auto& entryPoints = findResult->second.entryPoints;
+            auto* debloatedShader = findResult->second.debloatedShaders;
```

Comment:
no you need to go by reference `auto&` because right now you're making a temporary variable of pointer type, and its not getting set